botorch/sampling/index_sampler.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+r"""
+Sampler to be used with `EnsemblePosteriors` to enable
+deterministic optimization of acquisition functions with ensemble models.
+"""
+
+import torch
+from botorch.posteriors import Posterior
+from botorch.sampling.base import MCSampler
+from torch import Tensor
+
+
+class IndexSampler(MCSampler):
+    r"""A sampler that calls `posterior.rsample_from_base_samples` to
+    generate the samples via index base samples."""
+
+    def forward(self, posterior: Posterior) -> Tensor:
```

Comment:
Could type this as `EnsemblePosterior`.
```suggestion
    def forward(self, posterior: EnsemblePosterior) -> Tensor:
```